flink-connector-base/src/main/java/com/ververica/cdc/connectors/base/source/dialect/Dialect.java

```diff
@@ -0,0 +1,130 @@
+import org.apache.flink.table.types.logical.RowType;
+
+import com.ververica.cdc.connectors.base.source.internal.converter.JdbcSourceRecordConverter;
+import org.apache.flink.util.FlinkRuntimeException;
+import org.apache.kafka.connect.source.SourceRecord;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.sql.SQLException;
+import java.util.Map;
+
+import static org.apache.flink.table.api.DataTypes.FIELD;
+import static org.apache.flink.table.api.DataTypes.ROW;
+
+/** Handle the SQL dialect of jdbc driver. */
+@Internal
+public interface Dialect {
+	Logger LOG = LoggerFactory.getLogger(Dialect.class);
+
+	/**
+	 * Get  the name of dialect.
+	 *
+	 * @return the dialect name.
+	 */
+	String getName();
+
+	/**
+	 * Get converter that is used to convert JDBC object to {@link SourceRecord}.
+	 *
+	 * @param rowType the given row type.
+	 * @return a SourceRecord converter for the database.
+	 */
+	JdbcSourceRecordConverter getSourceRecordConverter(RowType rowType);
+
+	/**
+	 * Get a connection pool factory to create pooled DataSource.
+	 *
+	 * @return a connection pool factory.
+	 */
+	PooledDataSourceFactory getPooledDataSourceFactory();
+
+	/**
+	 * Creates and opens a new {@link JdbcConnection} backing connection pool.
+	 *
+	 * @param sourceConfig a basic source configuration.
+	 * @return a utility that simplifies using a JDBC connection.
+	 */
+	default JdbcConnection openJdbcConnection(SourceConfig sourceConfig) {
+		JdbcConnection jdbc =
+				new JdbcConnection(
+						sourceConfig.getDbzConfiguration(),
+						new JdbcConnectionFactory(
+								sourceConfig,
+								getPooledDataSourceFactory()));
+		try {
+			jdbc.connect();
+		} catch (Exception e) {
+			LOG.error("Failed to open database connection", e);
+			throw new FlinkRuntimeException(e);
+		}
+		return jdbc;
+	}
+
+	/**
+	 * Checks whether split column is evenly distributed across its range.
+	 *
+ 	 * @param splitColumn split column.
+	 * @return true that means split column with type BIGINT, INT, DECIMAL.
+	 */
+	default boolean isEvenlySplitColumn(Column splitColumn) {
+		DataType flinkType = fromDbzColumn(splitColumn);
+		LogicalTypeRoot typeRoot = flinkType.getLogicalType().getTypeRoot();
+
+		// currently, we only support the optimization that split column with type BIGINT, INT,
+		// DECIMAL
+		return typeRoot == LogicalTypeRoot.BIGINT
+				|| typeRoot == LogicalTypeRoot.INTEGER
+				|| typeRoot == LogicalTypeRoot.DECIMAL;
+	}
+
+	/**
+	 * convert dbz column to Flink row type.
+	 *
+	 * @param splitColumn split column.
+	 * @return flink row type.
+	 */
+	default RowType getSplitType(Column splitColumn) {
+		return (RowType)
+				ROW(FIELD(splitColumn.name(), fromDbzColumn(splitColumn)))
+						.getLogicalType();
+	}
+
+	/**
+	 * Get a corresponding Flink data type from a debezium {@link Column}
+	 *
+	 * @param splitColumn dbz split column.
+	 * @return flink data type
+	 */
+	DataType fromDbzColumn(Column splitColumn);
```

Comment:
This method looks strange in interface, Not every connector depends on debezium, right ?